cli_client/python/timesketch_cli_client/commands/search.py

```diff
@@ -0,0 +1,210 @@
+    elif output_format == 'tabular':
+        if show_headers:
+            result = tabulate(
+                dataframe, headers='keys', tablefmt='psql', showindex=False)
+        else:
+            result = tabulate(dataframe, tablefmt='psql', showindex=False)
+
+    return result
+
+
+def describe_query(search_obj):
+    """Print details of a search query nd filter.
+
+    Args:
+        search_obj: API Search object.
+    """
+    filter_pretty = json.dumps(search_obj.query_filter, indent=2)
+    click.echo(f'Query string: {search_obj.query_string}')
+    click.echo(f'Return fields: {search_obj.return_fields}')
+    click.echo(f'Filter: {filter_pretty}')
+
+
+@click.command('search')
+@click.option(
+    '--query', '-q', default='*',
+    help='Search query in Elasticsearch query string format')
+@click.option(
+    '--time', 'times', multiple=True,
+    help='Datetime filter (e.g. 2020-01-01T12:00)')
+@click.option(
+    '--time-range', 'time_ranges', multiple=True, nargs=2,
+    help='Datetime range filter (e.g: 2020-01-01 2020-02-01)')
+@click.option(
+    '--label', 'labels', multiple=True,
+    help='Filter events with label')
+@click.option(
+    '--header/--no-header', default=True,
+    help='Toggle header information (default is to show)')
+@click.option(
+    '--output-format', 'output',
+    help='Set output format (overrides global setting)')
+@click.option(
+    '--return-fields', 'return_fields', default='',
+    help='What event fields to show')
+@click.option(
+    '--order', default='asc',
+    help='Order the output (asc/desc) based on the time field')
+@click.option(
+    '--limit', type=int, default=40,
+    help='Limit amount of events to show (default: 40)')
+@click.option(
+    '--saved-search', type=int, help='Query and filter from saved search')
+@click.option(
+    '--describe', is_flag=True, default=False,
+    help='Show the query and filter then exit')
+@click.pass_context
+# pylint: disable=too-many-arguments
+def search_group(ctx, query, times, time_ranges, labels, header, output,
+                 return_fields, order, limit, saved_search, describe):
+    """Search and explore."""
+    sketch = ctx.obj.sketch
+    output_format = ctx.obj.output_format
+    search_obj = search.Search(sketch=sketch)
+
+    if output:
+        output_format = output
+
+    new_line = True
+    if output_format == 'csv':
+        new_line = False
+
+    # Construct query from saved search and return early.
+    if saved_search:
+        search_obj.from_saved(saved_search)
+        if describe:
+            describe_query(search_obj)
+            return
+        click.echo(format_output(
+            search_obj, output_format, header), nl=new_line)
+        return
+
+    # Construct the query from flags.
+    # TODO (berggren): Add support for query DSL.
+    search_obj.query_string = query
+    search_obj.return_fields = return_fields
+    search_obj.max_entries = limit
```

Comment:
I would opt to check if ou are setting these values... eg

```python
if return_fields:
    search_obj.return_fields = return_fields
```

that is at least for the `extra` fields, not for the query one since that is required, otherwise you may be overwriting some default values in the search object.

Reply:
this has not been addressed

Reply:
done